src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -479,6 +482,18 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       // Proceed under the assumption that this is a functional component
       workInProgress.tag = FunctionalComponent;
       if (__DEV__) {
+        const Component = workInProgress.type;
+
+        if (Component) {
+          warning(
+            !Component.childContextTypes,
+            '%s(...): childContextTypes cannot be defined on a functional component.',
+            Component.displayName || Component.name || 'Component'
+          );
+          if (Component.childContextTypes) {
+            warnAboutMissingGetChildContext(workInProgress);
```

Comment:
I think the test is wrong here. It doesn't make sense to both warn for `childContextTypes` being defined and also warn for that you should provide a `getChildContext` which you cannot do. It's not actionable.

IMO we should either fix this in Stack and remove it or gate the test on `ReactDOMFeatureFlags.useFiber` and change the assertion for the Fiber case.